packages/react-router/__tests__/useNavigate-test.tsx

```diff
@@ -344,4 +347,1343 @@ describe("useNavigate", () => {
       `);
     });
   });
+
+  describe("when relative navigation is handled via React Context", () => {
```

Comment:
These mimic the tests in `<Navigate>` and we have a set of them for both `<MemoryRouter>` and `<RouterProvider>` using `useNavigate`.  Then we test that `useNavigate` in `<MemoryRouter>` is unstable while it is stable in `<RouterProvider>`